src/trade-params/saga/DurationSaga.js

```diff
@@ -142,7 +142,9 @@ export function* handleStartTimeChange(action) {
     const contractPerType = contractNeeded[tradeCategory][type];
     const durationAllowed = allTimeRelatedFieldValid(newDateStart, duration, durationUnit, contractPerType);
 
-    if (durationAllowed) {
+     if (isValidTime(time)) {
+        yield put(updateTradeError(index, 'durationError', 'Time format is wrong!'));
```

Comment:
It's better to have

```
if (xxx) {
   ...
   return
}

if (yyyy) {
   ...
   return
}
```

than to have

```
if (xxx) {
   ...
   return
} else if (yyyy) {
   ...
   return
}
...
```

The first one is generally better, it tells reader that code exit with that condition, and that's all it does, 2nd chunk however requires reader to go until the end of code to know what's going on

I might have make the mistake at the first place, would you help me to fix it altogether?
Thanks